api/v1beta2/network_types.go

```diff
@@ -94,12 +104,57 @@ var (
 
 // TargetGroupHealthCheck defines health check settings for the target group.
 type TargetGroupHealthCheck struct {
-	Protocol        *string `json:"protocol,omitempty"`
-	Path            *string `json:"path,omitempty"`
-	Port            *string `json:"port,omitempty"`
-	IntervalSeconds *int64  `json:"intervalSeconds,omitempty"`
-	TimeoutSeconds  *int64  `json:"timeoutSeconds,omitempty"`
-	ThresholdCount  *int64  `json:"thresholdCount,omitempty"`
+	// The protocol to use to health check connect with the target.
+	// Defaults to TCP.
+	// +kubebuilder:default=TCP
+	// +kubebuilder:validation:Enum=TCP;HTTP;HTTPS
+	// +optional
```

Comment:
How does defaulting the field will interact with existing entries that don't have a value? I'm not super familiar with controller-gen, but it [looks like](https://github.com/kubernetes-sigs/controller-tools/blob/020c71865aa9bf42a25e0cf3cf0dedbf435694dd/pkg/crd/markers/validation.go#L485) the default value is written to the Schema, which would then be applied to objects where the field is empty.

Reply:
Yes, that would resolve my concern.